builtin/providers/circonus/resource_circonus_check_http.go

```diff
@@ -0,0 +1,381 @@
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateRegexp(checkHTTPBodyRegexpAttr, `.+`),
+			},
+			checkHTTPCAChainAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateRegexp(checkHTTPCAChainAttr, `.+`),
+			},
+			checkHTTPCertFileAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateRegexp(checkHTTPCertFileAttr, `.+`),
+			},
+			checkHTTPCiphersAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateRegexp(checkHTTPCiphersAttr, `.+`),
+			},
+			checkHTTPCodeRegexpAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      defaultCheckHTTPCodeRegexp,
+				ValidateFunc: validateRegexp(checkHTTPCodeRegexpAttr, `.+`),
+			},
+			checkHTTPExtractAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateRegexp(checkHTTPExtractAttr, `.+`),
+			},
+			checkHTTPHeadersAttr: &schema.Schema{
+				Type:         schema.TypeMap,
+				Elem:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateHTTPHeaders,
+			},
+			checkHTTPKeyFileAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateRegexp(checkHTTPKeyFileAttr, `.+`),
+			},
+			checkHTTPMethodAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      defaultCheckHTTPMethod,
+				ValidateFunc: validateRegexp(checkHTTPMethodAttr, `\S+`),
+			},
+			checkHTTPPayloadAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateRegexp(checkHTTPPayloadAttr, `\S+`),
+			},
+			checkHTTPReadLimitAttr: &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				ValidateFunc: validateFuncs(
+					validateIntMin(checkHTTPReadLimitAttr, 0),
+				),
+			},
+			checkHTTPURLAttr: &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validateFuncs(
+					validateHTTPURL(checkHTTPURLAttr, urlIsAbs),
+				),
+			},
+			checkHTTPVersionAttr: &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      defaultCheckHTTPVersion,
+				ValidateFunc: validateStringIn(checkHTTPVersionAttr, supportedHTTPVersions),
+			},
+		}, checkHTTPDescriptions),
+	},
+}
+
+// checkAPIToStateHTTP reads the Config data out of circonusCheck.CheckBundle into the
+// statefile.
+func checkAPIToStateHTTP(c *circonusCheck, d *schema.ResourceData) error {
+	httpConfig := make(map[string]interface{}, len(c.Config))
+
+	// swamp is a sanity check: it must be empty by the time this method returns
+	swamp := make(map[config.Key]string, len(c.Config))
+	for k, v := range c.Config {
+		swamp[k] = v
+	}
+
+	saveStringConfigToState := func(apiKey config.Key, attrName schemaAttr) {
```

Comment:
This closure can be made into a function if necessary but it felt like the code was easier to read with a closure vs a helper function with an opaque use.

As for what the closure does, it stages a value into the map that is saved into the state store.  The purpose of this closure is to catch any unsupported or unanticipated drift in the configuration block that comes back from the API vs what is supported by the resource.

The `config` block is a json hash of arbitrary key/value pairs that changes based on the check type.

LMK.